### ISSUE TITLE
Finalize Kube items

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#2233]
Unmerged/open issues upstream:

- [ ] GoogleCloudPlatform/kubernetes#2630
- [x] GoogleCloudPlatform/kubernetes#5093
- [ ] GoogleCloudPlatform/kubernetes#7372
- [ ] GoogleCloudPlatform/kubernetes#7893
- [x] GoogleCloudPlatform/kubernetes#8119
- [x] GoogleCloudPlatform/kubernetes#8203
- [x] GoogleCloudPlatform/kubernetes#8607
- [ ] GoogleCloudPlatform/kubernetes#8890
- [x] GoogleCloudPlatform/kubernetes#9384

**Comment by derekwaynecarr** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/7868

Needed to time box rogue build pods and deployments from running "forever".

**Comment by pweil-** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/7893

Required to provide security context constraints at a cluster level to allow definition of UID allocation, SELinux labels, service account permissions, and user permissions.

https://github.com/GoogleCloudPlatform/kubernetes/pull/8203

Supporting PR to allow admission controllers to be aware of `UserInfo`

**Comment by ironcladlou** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/7868 is also the foundation of user-initiated depoyment cancellation.

**Comment by brenton** [comment]
It would be great to have this fixed: https://github.com/GoogleCloudPlatform/kubernetes/issues/7370

The short of it is that because there's no locking or retrying around the iptables initialization code there is a case that can happen on OpenShift startup where the iptables rules for the kube-proxy will never be initialized.  You'll typically only see this if you are running the SDN and launch it the same time the kube-proxy starts up.  It's easy to reproduce by grabbing the lock with gdb while the kube-proxy tries to initialize itself.  Once it fails it will never reinitialize the rules.

**Comment by liggitt** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/8257

We'll want this so we can run the controller to ensure two service accounts exist per namespace:
* "default", with some read-only level of access, maybe to services/endpoints/imageStreams in the namespace, etc.
* "builder", with additional write access to imageStreams in the namespace

**Comment by csrwng** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/8233

If we allow you to exec to pods in your project, we should not let you exec into build pods (or anything else that's privileged) that basically give you root access to the node.

**Comment by smarterclayton** [comment]
We need more than this for 3.0 though - we need this to be an escalation check, right?  Just want to clarify that we don't physically need this change - we need the other one.

----- Original Message -----

**Comment by smarterclayton** [comment]
Brenton pleas spawn an issue in our repo for the proxy lock, I'll port my upstream change and add the loop

**Comment by deads2k** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/8072
https://github.com/GoogleCloudPlatform/kubernetes/pull/7974
https://github.com/GoogleCloudPlatform/kubernetes/pull/7966

All of these are related to attaching image pull secrets to pods.  Once these settle, there will be follow-on for pull secrets in service accounts.

**Comment by brenton** [comment]
@smarterclayton, I just opened https://github.com/openshift/origin/issues/2266.  I set the priority to P0 but feel free to change it if that's not right.

**Comment by derekwaynecarr** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/7372

Needed to ensure that we do not have dangling folders for each resource type after namespace deletion.  Enables ability to delete entire collection in etcd for a namespace.

**Comment by pmorie** [comment]
I am trying to get https://github.com/GoogleCloudPlatform/kubernetes/pull/7936 and related changes in but we may need to carry it as a patch for 1 release.

**Comment by deads2k** [comment]
pull secrets on service accounts: https://github.com/GoogleCloudPlatform/kubernetes/pull/8582

**Comment by timothysc** [comment]
perf regression - https://github.com/GoogleCloudPlatform/kubernetes/issues/8119

**Comment by deads2k** [comment]
still waiting on GoogleCloudPlatform/kubernetes#8582.  Supposedly I'm just waiting on a few other in the merge queue ahead of me.

**Comment by pmorie** [comment]
Env expansion: have acheived LGTM status on https://github.com/GoogleCloudPlatform/kubernetes/pull/8149 but awaiting merge

**Comment by pweil-** [comment]
Security Context Constraints update:

https://github.com/GoogleCloudPlatform/kubernetes/pull/7893 is expected to be carried as a patch, we have not had adequate feedback for this to be merged

https://github.com/GoogleCloudPlatform/kubernetes/pull/8203 has been merged into upstream

https://github.com/GoogleCloudPlatform/kubernetes/pull/8607 is required by SCC but is already in OpenShift as an upstream patch with https://github.com/openshift/origin/pull/2345

**Comment by markturansky** [comment]
These two PRs fix the CanSupport func for most PersistentVolume implementations.  The volume plugins won't work in PV without them.

https://github.com/GoogleCloudPlatform/kubernetes/pull/8688
https://github.com/GoogleCloudPlatform/kubernetes/pull/8732


PV Recycler (basic "rm -rf" scrub of NFS volumes):

https://github.com/GoogleCloudPlatform/kubernetes/pull/8334

**Comment by pmorie** [comment]
Found a nit that needs to be fixed in https://github.com/GoogleCloudPlatform/kubernetes/pull/8149 and self-unLGTM'd.  hopefully this will get in tomorrow.

**Comment by pmorie** [comment]
Lgtm status re-acheived

**Comment by pmorie** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/8149 is merged
https://github.com/GoogleCloudPlatform/kubernetes/pull/8582 is merged

**Comment by markturansky** [comment]
GoogleCloudPlatform/kubernetes#8688 merged
GoogleCloudPlatform/kubernetes#8732 merged

**Comment by ncdc** [comment]
Add support for pluggable Docker exec handlers: https://github.com/GoogleCloudPlatform/kubernetes/pull/8864

**Comment by derekwaynecarr** [comment]
Stop setting contianer.Requests since not supported in 1.0 as part of LimitRanger:
https://github.com/GoogleCloudPlatform/kubernetes/pull/8875

**Comment by abhgupta** [comment]
Allowing ActiveDeadlineSeconds to be updated for a pod: https://github.com/GoogleCloudPlatform/kubernetes/pull/8890

Added UPSTREAM patch in the meantime: https://github.com/openshift/origin/pull/2546

**Comment by derekwaynecarr** [comment]
Fix latency conditions in NamespaceExists admission plugin:
https://github.com/GoogleCloudPlatform/kubernetes/pull/8696

**Comment by derekwaynecarr** [comment]
Fix bug in quota describe output:
https://github.com/GoogleCloudPlatform/kubernetes/pull/9099

**Comment by markturansky** [comment]
Just a few.

https://github.com/GoogleCloudPlatform/kubernetes/pull/9069 (bug fix)
https://github.com/GoogleCloudPlatform/kubernetes/pull/9018 (PV Recycler)
https://github.com/GoogleCloudPlatform/kubernetes/pull/9020 (PV Recycler)
https://github.com/GoogleCloudPlatform/kubernetes/pull/9021 (PV Recycler)
https://github.com/GoogleCloudPlatform/kubernetes/pull/9024 (PV Recycler)

**Comment by smarterclayton** [comment]
At this point I'm not planning on doing another rebase.  If this is a problem for you let me know right now.

**Comment by smarterclayton** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/issues/2630 is must have fixed

**Comment by smarterclayton** [comment]
@pmorie please take point on 2630 (even if someone else fixes).

**Comment by derekwaynecarr** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/9235

Racing between delete controllers could cause project deletion to take O(minutes)
As part of this, need to update Origin namespace controller to also ignore item not found errors.

**Comment by derekwaynecarr** [comment]
Show resource limits in CLI
https://github.com/GoogleCloudPlatform/kubernetes/pull/9304

**Comment by smarterclayton** [comment]
Everyone on this list - you need to be getting your UPSTREAMS in to OpenShift master soon.  Be sure to document your upstream patches well so that the next time we do a rebase I can decode what the causes are.

----- Original Message -----

**Comment by ncdc** [comment]
Pluggable exec is in.

**Comment by pweil-** [comment]
Submitted https://github.com/openshift/origin/pull/2856 for https://github.com/GoogleCloudPlatform/kubernetes/pull/7893

**Comment by pmorie** [comment]
We need an upstream patch for https://github.com/GoogleCloudPlatform/kubernetes/pull/5093.

**Comment by derekwaynecarr** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/issues/8814

**Comment by derekwaynecarr** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/9358

**Comment by sdminonne** [comment]
Just added https://github.com/openshift/origin/pull/2917 for https://github.com/GoogleCloudPlatform/kubernetes/pull/5093

**Comment by derekwaynecarr** [comment]
Clean up of get pods output to be readable.
https://github.com/GoogleCloudPlatform/kubernetes/pull/9032

**Comment by mnagy** [comment]
Without https://github.com/GoogleCloudPlatform/kubernetes/pull/8688 and https://github.com/GoogleCloudPlatform/kubernetes/pull/8732, PVCs are not working properly. @markturansky @smarterclayton is there a plan to cherry pick these?

**Comment by markturansky** [comment]
@mnagy on it.

**Comment by markturansky** [comment]
@mnagy confirmed #8688 is already in OpenShift.  Please tell me otherwise if you can't run NFS as a PV.

**Comment by mnagy** [comment]
@markturansky I didn't try, but I can't see the `spec.PersistentVolumeSource.NFS != nil` condition from #8688 in here: https://github.com/openshift/origin/blob/master/Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/volume/nfs/nfs.go#L55

**Comment by markturansky** [comment]
@mnagy i'm crossing too many streams.  You are right and I made a PR to add to OS. #3002.

**Comment by pmorie** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/9384

**Comment by derekwaynecarr** [comment]
Disable pretty printing
https://github.com/GoogleCloudPlatform/kubernetes/pull/9361

**Comment by smarterclayton** [comment]
Are all the ports done here?  Please consolidate your lists and mark merged items as complete.  Need to know the remaining items.

**Comment by liggitt** [comment]
are you wanting to know whether they've merged to origin or to kube upstream (or both)?

**Comment by smarterclayton** [comment]
Only origin matters right now

**Comment by derekwaynecarr** [comment]
May also be related to scheduler issues I was seeing in scale testing.

https://github.com/GoogleCloudPlatform/kubernetes/pull/9613

**Comment by smarterclayton** [comment]
Let's cherry pick this - we use a lot more run once pods.

**Comment by derekwaynecarr** [comment]
https://github.com/GoogleCloudPlatform/kubernetes/pull/9765

**Comment by derekwaynecarr** [comment]
Reducing load against docker is a good thing to do.  https://github.com/GoogleCloudPlatform/kubernetes/pull/9809

**Comment by smarterclayton** [comment]
Whichever of us gets to this tomorrow should grab both of those.  May be the last we get.

**Comment by ncdc** [comment]
> Reducing load against docker is a good thing to do

Except for the aforementioned fact that the runtime cache hammers Docker with 20 list containers calls every 10 seconds.

**Comment by smarterclayton** [comment]
But the runtime cache is already in use, so you're paying that cost already.

**Comment by smarterclayton** [comment]
We're out of time.  Anything remaining should be spawned as issues if critical and we'll triage individually.

**Comment by liggitt** [comment]
Swept through and marked merged/resolved issues we've picked up fixes for. Noted open issues in the PR description

**Comment by ncdc** [comment]
Closing. Will start a new issue to track the next batch.
